crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,490 @@
+            expected_fee_asset,
+        } = self;
+
+        let sequencer_cometbft_client =
+            sequencer_client::HttpClient::new(&*sequencer_cometbft_endpoint)
+                .wrap_err("failed constructing cometbft http client")?;
+
+        Ok(Startup {
+            shutdown_token,
+            state,
+            sequencer_chain_id,
+            sequencer_cometbft_client,
+            sequencer_bridge_address,
+            expected_fee_asset,
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize)]
+pub(super) struct Info {
+    pub(super) starting_rollup_height: u64,
+    pub(super) fee_asset: asset::Denom,
+    pub(super) chain_id: String,
+}
+
+#[derive(Debug, Clone)]
+pub(super) struct InfoHandle {
+    rx: watch::Receiver<state::StateSnapshot>,
+}
+
+impl InfoHandle {
+    pub(super) fn new(rx: watch::Receiver<state::StateSnapshot>) -> Self {
+        Self {
+            rx,
+        }
+    }
+
+    pub(super) async fn get_info(&mut self) -> eyre::Result<Info> {
+        let state = self
+            .rx
+            .wait_for(|state| state.get_startup_info().is_some())
+            .await
+            .wrap_err("failed to get startup info")?;
+
+        Ok(state
+            .get_startup_info()
+            .expect("the previous line guarantes that the state is intialized")
+            .clone())
+    }
+}
+
+pub(super) struct Startup {
+    shutdown_token: CancellationToken,
+    state: Arc<State>,
+    sequencer_chain_id: String,
+    sequencer_cometbft_client: sequencer_client::HttpClient,
+    sequencer_bridge_address: Address,
+    expected_fee_asset: asset::Denom,
+}
+
+impl Startup {
+    pub(super) async fn run(mut self) -> eyre::Result<()> {
+        let shutdown_token = self.shutdown_token.clone();
+
+        let startup_task = tokio::spawn({
+            let state = self.state.clone();
+            async move {
+                self.confirm_sequencer_config()
+                    .await
+                    .wrap_err("failed to confirm sequencer config")?;
+                let starting_rollup_height = self
+                    .get_starting_rollup_height()
+                    .await
+                    .wrap_err("failed to get next rollup block height")?;
+
+                // send the startup info to the submitter
+                let info = Info {
+                    chain_id: self.sequencer_chain_id.clone(),
+                    fee_asset: self.expected_fee_asset,
+                    starting_rollup_height,
+                };
+
+                state.set_startup_info(info);
+
+                Ok(())
+            }
+        });
+
+        tokio::select!(
+            () = shutdown_token.cancelled() => {
+                Err(eyre!("startup was cancelled"))
+            }
+            res = startup_task => {
+                match res {
+                    Ok(Ok(())) => Ok(()),
+                    Ok(Err(err)) => {
+                        error!(%err, "startup failed");
```

Comment:
This log is IMO not necesary. This will be printed one level above again I think.